examples/transit_vnet_common/example.tfvars

```
@@ -1,75 +1,207 @@
-    network_security_group = "sg-mgmt"
-  }
-  "subnet-private" = {
-    address_prefixes       = ["10.110.0.0/24"]
-    network_security_group = "sg-private"
-    route_table            = "private_route_table"
-  }
-  "subnet-public" = {
-    address_prefixes       = ["10.110.129.0/24"]
-    network_security_group = "sg-public"
-  }
-}
 
-frontend_ips = {
-  "frontend01" = {
-    create_public_ip = true
-    rules = {
-      "balancessh" = {
-        protocol = "Tcp"
-        port     = 22
+
+# --- LOAD BALANCING PART --- #
+load_balancers = {
+  "lb-public" = {
+    vnet_name                         = "transit-vnet"
+    network_security_group_name       = "public"
+    network_security_allow_source_ips = ["y.y.y.y"] # TODO adjust to the public IPs that will connect to the public Load Balancer
+
+    frontend_ips = {
+      "palo-lb-app1-pip" = { # TODO this is just a basic load balancing rule that will balance HTTP(s) traffic, add more rules to balance different types of traffic
+        create_public_ip = true
+        rules = {
+          "balanceHttp" = {
+            protocol = "Tcp"
+            port     = 80
+          }
+          "balanceHttps" = {
+            protocol = "Tcp"
+            port     = 443
+          }
+        }
+      }
+    }
+  }
+  "lb-private" = {
+    frontend_ips = {
+      "ha-ports" = {
+        vnet_name          = "transit-vnet"
+        subnet_name        = "private"
+        private_ip_address = "10.0.0.50"
+        rules = {
+          HA_PORTS = {
+            port     = 0
+            protocol = "All"
+          }
+        }
       }
     }
   }
 }
 
-vmseries = {
-  "fw00" = { avzone = 1 }
-  "fw01" = { avzone = 2 }
-}
 
-common_vmseries_version = "9.1.3"
-common_vmseries_sku     = "bundle1"
-storage_account_name    = "pantfstorage"
-storage_share_name      = "bootstrapshare"
 
-files = {
-  "files/authcodes"    = "license/authcodes" # authcode is required only with common_vmseries_sku = "byol"
-  "files/init-cfg.txt" = "config/init-cfg.txt"
+# --- VMSERIES PART --- #
+availability_set = {
+  "vmseries" = {}
 }
 
-avzones = ["1", "2", "3"]
\ No newline at end of file
+vmseries_version = "10.2.2"
+vmseries_vm_size = "Standard_DS3_v2"
+vmseries_sku     = "byol"
+# vmseries_password = "" # TODO by default the VM-Series admin password is autogenerated, uncomment and provide you own
+vmseries = {
+  "vmseries-1" = {
+    availability_set_name = "vmseries"
+    app_insights_settings = {}
+    bootstrap_options     = "type=dhcp-client" # TODO add licensing, panorama configuration if needed
+    vnet_name             = "transit-vnet"
+    interfaces = [
+      {
+        name               = "mgmt"
+        subnet_name        = "management"
+        private_ip_address = "10.0.0.10"
+        create_pip         = true
+      },
+      {
+        name                 = "private"
+        subnet_name          = "private"
+        backend_pool_lb_name = "lb-private"
```

For clarity, let's use `backend_pool_name` instead. Otherwise, the user needs to have prior knowledge that [we are creating a backend pool with the load balancer name](https://github.com/PaloAltoNetworks/terraform-azurerm-vmseries-modules/blob/9dfe470572d5b7b2e53cb8a30c43a116ea784870/modules/loadbalancer/main.tf#L119) and introduce a `backend_pool_name` after line #106

Not related to this, but please fix the [variable description](https://github.com/PaloAltoNetworks/terraform-azurerm-vmseries-modules/blob/9dfe470572d5b7b2e53cb8a30c43a116ea784870/modules/loadbalancer/variables.tf#L154-L156) to indicate that we are defaulting to load balancer name if a name is not provided for the backend.

---

well, you're right, this name is confusing, but this is not the backend name, it's actually the load balancer name. If you look [here](https://github.com/PaloAltoNetworks/terraform-azurerm-vmseries-modules/blob/228-examples-refactor-transit_vnet_common/examples/transit_vnet_common/main.tf#L133) you'll see that we are using the `backend_pool_lb_name` property to target a specific load balancer and then we just access it's backend pool ID. Backend name in this situation i never used.

maybe we should change it to something like:

```suggestion
        load_balancer_name = "lb-private"
```